### ISSUE TITLE
Assert chosen mode exists

### ISSUE THREAD

**Issue by dmed256** [libocca/occa#232]
Currently `Serial` mode is used by default if the chosen mode is not enabled

```
[CUDA] mode is not enabled, defaulting to [Serial] mode
```

Switch to throwing an `occa::exception` instead and allow users to use

```
occa::modeIsEnabled("CUDA")
```

for manual checking

**Comment by suniljoshi82** [comment]
I could not get the "manual checking" bit, is the code below good enough?

  mode_v* getMode(const occa::properties &props) {
    std::string mode = props["mode"];
    if (!mode.size()) {
    	throw occa::exception("OCCA modes", __FILE__, "getMode", __LINE__, "No OCCA mode given");
    } else if (!modeIsEnabled(mode)) {
    	throw occa::exception("OCCA modes", __FILE__, "getMode", __LINE__, "[" + mode + "] mode is not enabled");
    }
    return modeMap()[mode];
  }

**Comment by dmed256** [comment]
@suniljoshi82 Yeah, that looks good!

I would change the exceptions to use our `OCCA_ERROR` macros instead:

```cpp
  mode_v* getMode(const occa::properties &props) {
    const std::string mode = props["mode"];

    OCCA_ERROR("No OCCA mode given", mode.size() > 0);
    OCCA_ERROR("[" << mode << "] mode is not enabled", modeIsEnabled(mode));

    return modeMap()[mode];
  }
```

**Comment by dmed256** [comment]
Closed with #282
